### PR TITLE
Fixed double imports of data when using script window

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes double imports of data as outlined in issue #8509.
This PR also fixes optimises running of commands through the script window.

@africanmathsinitiative/developers this is ready for review.

@rdstern I'll keep this updated with the master branch. 
If possible, kindly do all possible tests on this branch in your daily operations. I'm particularly keen on which dialog scripts would fail when run through the script window.

@lloyddewit as mentioned to me by @rdstern you may be working on this as well.

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@Patowhiz looks great, but with one problem.
It now gives an error running a line when there is a comment at the end of that line.  Here is the example - which runs without error on the master.  I have added an extra comment where there is now an error.  It can't now process a line where there is a command followed by a comment. 

```
 library(agridat)
  data(hanover.whitepine)
  dat <- hanover.whitepine
  data_book$import_data(data_tables=list(dat=dat))

  libs(lattice)
  ## Relatively high male-female interaction in growth comared
  ## to additive gene action. Response is more consistent within
  ## male progeny than female progeny.
  # with(dat, interaction.plot(female, male, length))
  # with(dat, interaction.plot(male, female, length))
  bwplot(length ~ male|female, data=dat,
         main="hanover.whitepine - length for male:female crosses",
         xlab="Male parent", ylab="Epicotyl length")
  
  # Progeny sums match Becker p 83
#This is the first line that now gives an error and therefore doesn't give the result.
  sum(dat$length) # 380.58

    aggregate(length ~  female + male, data=dat, FUN=sum)
  
  # Sum of squares matches Becker p 85
  m1 <- aov(length ~ rep + male + female + male:female, data=dat)
  anova(m1)
  
  # Variance components match Becker p. 85
  libs(lme4)
  libs(lucid)
  m2 <- lmer(length ~ (1|rep) + (1|male) + (1|female) + (1|male:female), data=dat)
  #as.data.frame(lme4::VarCorr(m2))
  vc(m2)
  ##         grp        var1 var2    vcov  sdcor
  ## male:female (Intercept) <NA> 0.1369  0.3699
  ##      female (Intercept) <NA> 0.02094 0.1447
  ##        male (Intercept) <NA> 0.1204  0.3469
  ##         rep (Intercept) <NA> 0.01453 0.1205
  ##    Residual        <NA> <NA> 0.2004  0.4477
  
  # Becker used this value for variability between individuals, within plot
  s2w <- 1.109
  
  # Calculating heritability for individual trees
  s2m <- .120
  s2f <- .0209
  s2mf <- .137
#patrick, the next 4 lines now each give an error, and therefor not the result.
  vp <- s2m + s2f + s2mf + s2w  # variability of phenotypes = 1.3869
  4*s2m / vp # heritability male 0.346
  4*s2f / vp # heritability female 0.06
  2*(s2m+s2f)/vp # heritability male+female .203
  # As shown in the boxplot, heritability is stronger through the
  # males than through the females.
```

**Comment by Patowhiz** [comment]
@rdstern I have had to restore not running comments from the script window. Your examples above show why we need to be removing comments before running the scripts. I will fix showing comments in an output window in another PR. To show the comments, we don't have to run them, a feature that is currently not yet implemented.

**Review by rdstern — APPROVED** [review]
@Patowhiz the double data frames seems solved.
I am also happy that there can be a different way of putting the comments into the output window.  I found running them as commands menas that they have a separate little box per line, and that isn't needed.

I think @lloyddewit also plans to include this feature in his #8494 pull request.  That may be in a couple of weeks, but that would be fine.

File: instat/clsRLink.vb [hunk 1]
```diff
@@ -824,15 +824,8 @@ Public Class RLink
...
                 ElseIf iCallType = 5 Then
                     'else if script comes from script window
-                    Dim bSuccess As Boolean = Evaluate(strScript, bSilent:=bSilent, bSeparateThread:=bSeparateThread, bShowWaitDialogOverride:=bShowWaitDialogOverride)
-
-                    'if not an assignment operation, then capture the output
-                    If Not strScript.Contains("<-") AndAlso bSuccess Then
-                        Dim strScriptAsSingleLine As String = strScript.Replace(vbCrLf, String.Empty)
-                        strScriptAsSingleLine = strScriptAsSingleLine.Replace(vbCr, String.Empty)
-                        strScriptAsSingleLine = strScriptAsSingleLine.Replace(vbLf, String.Empty)
-                        strOutput = GetFileOutput("view_object_data(object = " & strScriptAsSingleLine & " , object_format = 'text' )", bSilent, bSeparateThread, bShowWaitDialogOverride)
```
Comment (lloyddewit) [inline instat/clsRLink.vb:834]: @Patowhiz If these lines are removed, then will multiline line statements fail?
```
3 *
6 /
2
```

**Comment by rdstern** [comment]
@lloyddewit I had a look again.  Slghtly surprised that my approval is still holding.  Anyway I checked again, and the double dta frame issue seems to be solved.  I think that the echoing of the comments is now waiting for you.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@Patowhiz @rdstern Thank you for the feedback. I'll merge this PR, and when @rdstern tests PR #8494, he can ensure that the tests he ran on this PR still pass.

**Pull request merged**